src/canvaslms/cli/discussions.nw

```diff
@@ -0,0 +1,227 @@
+  announce_parser = subp.add_parser("announce",
+    help="Post announcements to courses",
+    description="Post announcements to one or more courses matching a regex pattern. "
+                "The announcement can be provided via command line or interactively using an editor.")
+  announce_parser.set_defaults(func=announce_command)
+  <<add announce arguments>>
+
+def add_list_command(subp):
+  """Adds the list subcommand"""
+  list_parser = subp.add_parser("list",
+    help="List discussions and announcements",
+    description="List discussions or announcements from courses.")
+  list_parser.set_defaults(func=list_command)
+  <<add list arguments>>
+@
+
+
+\section{Announce command arguments}
+
+The announce command requires a title and supports various ways to specify the message.
+<<add announce arguments>>=
+announce_parser.add_argument("title",
+  help="Title of the announcement")
+announce_parser.add_argument("-c", "--course", required=True,
+  help="Regex matching courses on title, course code or Canvas ID")
+announce_parser.add_argument("-m", "--message",
+  help="Message content of the announcement (use -i for interactive mode)")
+announce_parser.add_argument("-i", "--interactive",
+  action="store_true", default=False,
+  help="Interactive mode: open editor to write announcement message in Markdown")
+@
+
+\section{List command arguments}
+
+The list command can list either announcements or discussions.
+<<add list arguments>>=
+list_parser.add_argument("type",
+  choices=["announcements", "discussions"],
+  help="Type of content to list")
+list_parser.add_argument("-c", "--course", required=True,
+  help="Regex matching courses on title, course code or Canvas ID")
+@
+
+
+\section{The announce command function}
+
+The main function that processes the announce command.
+<<functions>>=
+def announce_command(config, canvas, args):
+  """Posts announcements to matching courses"""
+  
+  # Get list of matching courses
+  course_list = list(canvaslms.cli.courses.filter_courses(canvas, args.course))
+  
+  if not course_list:
+    print(f"No courses found matching pattern: {args.course}", file=sys.stderr)
+    sys.exit(1)
+  
+  # Get the message content
+  if args.interactive:
+    message = get_message_from_editor()
+  elif args.message:
+    message = args.message
+  else:
+    print("Error: Either provide -m/--message or use -i/--interactive mode", file=sys.stderr)
+    sys.exit(1)
+  
+  if not message.strip():
+    print("Error: Message cannot be empty", file=sys.stderr)
+    sys.exit(1)
+  
+  # Show courses that will receive the announcement
+  print(f"Will post announcement '{args.title}' to {len(course_list)} course(s):", file=sys.stderr)
+  for course in course_list:
+    print(f"  - {course.course_code}: {course.name}", file=sys.stderr)
+  
+  # Confirm before posting
+  confirm = input("Continue? (y/N): ")
+  if confirm.lower() not in ['y', 'yes']:
+    print("Cancelled.", file=sys.stderr)
+    sys.exit(0)
```

Comment:
Only confirm interactively when run in interactive mode. We don't want any confirmation even everything is given on the command line without the --interactive option.